apis/placement/v1beta1/override_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Override defines a group of override policies about how to override the selected resources and how to apply these resources
+// to the target clusters.
+// Note: override may fail, and it will be reflected on the placement status.
+type Override struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ClusterResourcePlacement.
+	// +required
+	Spec OverrideSpec `json:"spec"`
+}
+
+// OverrideSpec defines the desired state of the Override.
+type OverrideSpec struct {
+	// If none of the ClusterResourceSelectors and ResourceSelectors are specified, it means selecting all resources.
+	// The ClusterResourceSelectors and ResourceSelectors are `ORed`.
+
+	// ClusterResourceSelectors is an array of selectors used to select cluster scoped resources. The selectors are `ORed`.
+	// If a namespace is selected, ALL the resources under the namespace are selected automatically.
+	// You can have 1-100 selectors.
+	// +optional
+	ClusterResourceSelectors []ClusterResourceSelector `json:"clusterResourceSelectors,omitempty"`
+
+	// ResourceSelectors is an array of selectors used to select namespace scoped resources. The selectors are `ORed`.
+	// You can have 1-100 selectors.
+	// +optional
+	ResourceSelectors []ResourceSelector `json:"resourceSelectors,omitempty"`
+
+	// ClusterSelectors selects the target clusters.
+	// The resources will be overridden before applying to the matching clusters.
+	// If ClusterSelector is not set, it means selecting ALL the member clusters.
+	// +optional
+	ClusterSelector *ClusterSelector `json:"clusterSelector,omitempty"`
+
+	// OverridePolices defines how to override the selected resources.
+	// +optional
+	OverridePolices *OverridePolices `json:"overridePolices,omitempty"`
+
+	// ApplyMode defines how to apply resources if the resources exist on the target clusters.
+	// It can be "CreateOnly", "ServerSideApply" or "ServerSideApplyWithForceConflicts". Default is CreateOnly.
+	// +kubebuilder:validation:Enum=CreateOnly;ServerSideApply;ServerSideApplyWithForceConflicts
+	// +kubebuilder:default=CreateOnly
+	// +optional
+	ApplyMode ApplyMode `json:"applyMode,omitempty"`
+
+	// Priority is an integer defining the relative importance of this override compared to others.
+	// Lower number is considered higher priority.
```

Comment:
any reason that lower means higher priority？

Reply:
In the azure networking world, ["Azure default security rules are given the highest number with the lowest priority to ensure that custom rules are always processed first."](https://learn.microsoft.com/en-us/azure/virtual-network/network-security-groups-overview#security-rules)

Similarly, [the azure firewall policy](https://learn.microsoft.com/en-us/azure/firewall/rule-processing) follows the same pattern. "Rules are processed based on Rule Collection Group Priority and Rule Collection priority. Priority is any number between 100 (highest priority) to 65,000 (lowest priority). Highest priority Rule Collection Groups are processed first. Inside a rule collection group, Rule Collections with highest priority (lowest number) are processed first."